compiler/semantic.ml

```diff
@@ -48,6 +51,21 @@ let check_binop_type (left_expr : data_type) (op : Ast.math_op) (right_expr : da
 	| (String, Add, String) -> String
 	| (_, _, _) -> raise (Failure "cannot perform binary operations with provided arguments")
 
+let check_bool_expr_binop_type (left_expr : data_type) (op : Ast.bool_op) (right_expr : data_type) = match op 
+		with Equal | Neq -> (match (left_expr, right_expr)
+			with (Int, Int) -> Int
+			| (Float, Float) -> Float
+			| (String, String) -> String
+			| (Bool, Bool) -> Bool
+			| _ -> raise (Failure "cannot perform binary operations with provided arguments")
```

Comment:
what about our wildcard type for data gotten from a json? maybe this is something we should talk about in more detail at our meeting, also